quil-rs/src/expression/mod.rs

```diff
@@ -1046,6 +1046,13 @@ mod tests {
             ("pi", Expression::Number(PI.into())),
             ("pi/2", Expression::Number((PI / 2.0).into())),
             ("pi * pi", Expression::Number((PI.powi(2)).into())),
+            (
+                "(a[0]*2*pi)/6.283185307179586",
```

Comment:
👍

Reply:
Does this also cover `2*pi*a[0]`? Also curious if this can be generalized into `a[0]*x/x`

Reply:
I think there may be a way to put that together with the current rules + the changes here, but maybe adding commutativity (& trimming the ruleset, see my previous comment) might make sense.

As for your `a[0]*x/x` example, I think the `"div-mul-2"` rule covers it as long as `x` is nonzero:

```
rw!("div-mul-2" ; "(/ (* ?b ?a) ?a)" => "?b" if is_not_zero("?a")),
```